lib/grinder_tools.dart

```diff
@@ -84,53 +85,57 @@ void runProcess(String executable,
       environment: environment);
 
   if (!quiet) {
-    if (result.stdout != null && !result.stdout.isEmpty) {
+    if (result.stdout != null && result.stdout.isNotEmpty) {
       log(result.stdout.trim());
     }
   }
 
-  if (result.stderr != null && !result.stderr.isEmpty) {
+  if (result.stderr != null && result.stderr.isNotEmpty) {
     log(result.stderr);
   }
 
   if (result.exitCode != 0) {
-    throw new GrinderException(
-        "${executable} failed with a return code of ${result.exitCode}");
+    throw new ProcessException(executable, result.exitCode, result.stderr);
```

Comment:
Should we also return the stdout with this exception? Some CLIs may not put all the info relevant to triaging a failure into stderr.